src/coreclr/jit/importer.cpp

```diff
@@ -13855,14 +13855,17 @@ unsigned Compiler::impInlineFetchLocal(unsigned lclNum DEBUGARG(const char* reas
 //
 GenTree* Compiler::impInlineFetchArg(InlArgInfo& argInfo, const InlLclVarInfo& lclInfo)
 {
+    GenTree* argNode = argInfo.arg->GetNode();
+    assert(!argNode->OperIs(GT_RET_EXPR));
+
     // Cache the relevant arg and lcl info for this argument.
     // We will modify argInfo but not lclVarInfo.
-    const bool      argCanBeModified = argInfo.argHasLdargaOp || argInfo.argHasStargOp;
-    const var_types lclTyp           = lclInfo.lclTypeInfo;
-    GenTree*        op1              = nullptr;
 
-    GenTree* argNode = argInfo.arg->GetNode();
-    assert(!argNode->OperIs(GT_RET_EXPR));
+    // Ignore Ldarga on CNS_STR, string literals can't be modified anyway (UB).
```

Comment:
Not sure I understand, `ldarga` does not take the address of the string contents, it takes the address of the string local. Even if it starts by pointing to a string literal it may be modified to point to something else through that pointer. Am I missing something?

Reply:
Oops, reverted - it had no impact on the example in the description and diffs.

As we discussed offline, this particular `ldarga` was caused by `ldarga.s + constrained callvirt` inside `Dictionary.FindValue` for `uint hashCode = (uint)key.GetHashCode();`